ksqldb-engine/src/main/java/io/confluent/ksql/util/QueryMetadataImpl.java

```diff
@@ -196,14 +196,14 @@ public void initialize() {
               Throwables.getStackTraceAsString(e),
               errorType
           );
-      listener.onError(this, queryError);
       queryErrors.add(queryError);
       LOG.error(
           "Unhandled exception caught in streams thread {}. ({})",
           Thread.currentThread().getName(),
           errorType,
           e
       );
+      listener.onError(this, queryError);
```

Comment:
Curious, why was it required to move this call?

Reply:
I was failing some tests in `SecureIntegrationTest` that called `assertQueryFailsWithUserError`. It seemed that the `queryError` was not being added and so the tests failed, but when i moved the listener to the end they passed

Reply:
We looked into this offline and this line can be reverted.